crates/rust-client/src/errors.rs

```diff
@@ -58,8 +58,8 @@ pub enum ClientError {
     AddNewAccountWithoutSeed,
     #[error("error with merkle path")]
     MerkleError(#[from] MerkleError),
-    #[error("the transaction didn't produce the expected notes corresponding to note ids")]
-    MissingOutputNotes(Vec<NoteId>),
+    #[error("the transaction didn't produce the expected output note recipient digests ({0:?})")]
```

Comment:
nit:
```suggestion
    #[error("the transaction didn't produce the output notes with the expected recipient digests ({0:?})")]
```
I think to someone external the relation between "own notes", "output notes", and "expected recipient" may not be super clear so it's worth being as explicit as possible IMO